ScnSoft-Education/LeetCodeTraining/ValidParentheses/ValidParenthesesSolution.cs

```diff
@@ -18,18 +18,18 @@ public static bool IsValid(string s)
                 if (parentheses.ContainsKey(s[i]))
                 {
                     st.Push(s[i]);
+                    continue;
                 }
-                else if (st.TryPeek(out char p) && parentheses[p] == s[i])
-                {
-                    st.Pop();
-                }
-                else
+
+                if (st.Any() && parentheses[st.Pop()] == s[i])
```

Comment:
I'm not sure, but st.TryPop() can be the best option here.

Reply:
hello @shomzascnsoft thank you for suggestion, this code works and it does indeed look like the most suitable option:
```
//if (st.Any() && parentheses[st.Pop()] == s[i])
if (st.TryPop(out char p) && parentheses[p] == s[i])
{
    continue;
}
```
I guess I should've used it from the very start, but for some reason I thought that TryPeek'ing and comparing the element before actually Popping it in this particular case was more correct. But apparently we can TryPop the element anyways and check the result, so I agree that TryPop works the best there.